controller/postgres/deletion_protection.go

```diff
@@ -0,0 +1,133 @@
+package postgres
+
+import (
+	"context"
+	"encoding/json"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	logging "sigs.k8s.io/controller-runtime/pkg/log"
+	"strconv"
+	"time"
+
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	finalizerName  = "appcat.io/deletionProtection"
+	currentTimeKey = "now"
```

Comment:
Nit: this seems to be only used for the tests, so maybe move it to the `deletion_protection_test.go` file.